checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,387 @@
+            final String methodName, final MethodInvocationTree tree) {
+        if (!useValueChecker) {
+            return methodName;
+        }
+
+        ExecutableElement invokedMethod = TreeUtils.elementFromUse(tree);
+        if (!"com.amazonaws.services.ec2.model.DescribeImagesRequest"
+                .equals(ElementUtils.enclosingClass(invokedMethod).getQualifiedName().toString())) {
+            return methodName;
+        }
+
+        if ("withFilters".equals(methodName) || "setFilters".equals(methodName)) {
+            ValueAnnotatedTypeFactory valueATF = getTypeFactoryOfSubchecker(ValueChecker.class);
+            for (Tree filterTree : tree.getArguments()) {
+                String adjustedMethodName = filterTreeToMethodName(filterTree, valueATF);
+                if (adjustedMethodName != null) {
+                    return adjustedMethodName;
+                }
+            }
+        }
+        return methodName;
+    }
+
+    /**
+     * Determine the name of the method in DescribeImagesRequest that is equivalent to the Filter in
+     * the given tree.
+     *
+     * <p>Returns null unless the argument is one of the following:
+     *
+     * <ul>
+     *   <li>a constructor invocation of the Filter constructor whose first argument is the name,
+     *       such as {@code new Filter("owner").*}, or
+     *   <li>a call to the withName method, such as {@code new Filter().*.withName("owner").*}.
+     * </ul>
+     *
+     * In those cases, it returns either the argument to the constructor or the argument to the last
+     * invocation of withName ("owner" in both of the above examples).
+     *
+     * @param filterTree the tree that represents the filter (an argument to the withFilters or
+     *     setFilters method)
+     * @param valueATF the type factory from the Value Checker
+     * @return the adjusted method name, or null if the method name should not be adjusted
+     */
+    private @Nullable String filterTreeToMethodName(
+            Tree filterTree, ValueAnnotatedTypeFactory valueATF) {
+        while (filterTree != null && filterTree.getKind() == Tree.Kind.METHOD_INVOCATION) {
+
+            MethodInvocationTree filterTreeAsMethodInvocation = (MethodInvocationTree) filterTree;
+            String filterMethodName = TreeUtils.methodName(filterTreeAsMethodInvocation).toString();
+            if ("withName".equals(filterMethodName)
+                    && filterTreeAsMethodInvocation.getArguments().size() >= 1) {
+                Tree withNameArgTree = filterTreeAsMethodInvocation.getArguments().get(0);
+                String withNameArg =
+                        ValueCheckerUtils.getExactStringValue(withNameArgTree, valueATF);
+                return filterKindToMethodName(withNameArg);
+            }
+
+            if (TreeUtils.isMethodInvocation(
+                    filterTree, collectionsSingletonList, getProcessingEnv())) {
+                // Descend into a call to Collections.singletonList()
```

Comment:
Descending into singletonList is only relevant at the top level (as the argument `filterTree`), so it doesn't belong within the `while` loop.

In fact, it doesn't belong in `filterTreeToMethodName`.  Move it into `adjustMethodNameUsingValueChecker`, since it is logically part of the `for (Tree filterTree : tree.getArguments())` loop.  I think this will clarify the logic.